src/test/java/com/hubspot/imap/MessageAddListenerTest.java

```diff
@@ -31,6 +34,19 @@ public void closeClient() throws Exception {
     client.close();
   }
 
+  @Test
+  public void testOnKeepAlive_doesCallMessageListener() throws Exception {
+    CountDownLatch countDownLatch = new CountDownLatch(1);
+
+    client.getState().onMessageAdd((o, n) -> countDownLatch.countDown());
+
+    CompletableFuture<OpenResponse> openFuture = client.open(DEFAULT_FOLDER, FolderOpenMode.READ);
+    openFuture.get(30, TimeUnit.SECONDS);
```

Comment:
OOC. Why do the futures have different time limits? Do some requests take longer than others? 30 seconds for open vs 10 seconds for fetch

Reply:
Just updated the PR to only call `get` with a timeout on methods that could possibly block forever if the NioImapClient implementation was bad. 🤷